core/src/main/java/com/klaviyo/coresdk/KlaviyoConfig.kt

```diff
@@ -0,0 +1,66 @@
+package com.klaviyo.coresdk
+
+import android.content.Context
+import java.lang.Exception
+
+class KlaviyoMissingAPIKeyException: Exception("You must declare an API key for the Klaviyo SDK")
+
+class KlaviyoMissingContextException: Exception("You must add your application context to the Klaviyo SDK")
+
+class KlaviyoConfig {
+    internal companion object {
+        private const val NETWORK_TIMEOUT_DEFAULT: Int = 500
+        private const val NETWORK_FLUSH_INTERVAL_DEFAULT: Int = 60000
+
+        lateinit var apiKey: String
+        lateinit var applicationContext: Context
+        var networkTimeout = NETWORK_TIMEOUT_DEFAULT
+        var networkFlushInterval = NETWORK_FLUSH_INTERVAL_DEFAULT
```

Comment:
oh so we're going with this approach for config now?

Reply:
Ya, as I worked on following tickets I hit limitations on what I had done before, so I'm turning this into an object class so that it's a singleton and just using the builder as a convenience tool to populating this singleton.
I know builder + singleton can equal code smell. The encapsulation isn't perfect here but its also pretty harmless and for the sake of at least getting some easy-to-configure vars built out fast, I think it's adequate for the time being

Reply:
I think in the future we will probably end up with a `Klaviyo` core class that the user calls `init` on and passes in a built out `KlaviyoConfig` instance, for storing config and kicking off internal threads and the like. But I'm tackling that piece of architecture in a later ticket